rfcs/055-prismic-api/README.md

```diff
@@ -0,0 +1,249 @@
+- The elasticsearch index mapping should represent the contract between the pipeline and the API. The API shouldn't need to know anything about the structure of the data in Prismic.
+- We won't include an endpoint for `/series`, nor will we include series nodes in the `/articles` endpoint. If an article is part of a series, we'll denormalise information from its parent onto the article itself to use for relevance and visual signposting in results.
+
+## Proposed endpoints
+
+### `/articles`
+
+#### Example response
+
+```http
+GET /articles/{id}
+{
+    "id": "",
+    "format": "",
+    "title": "",
+    "body": "", // not exposed by default, should require an ?includes=body
+    "standfirst": "",
+    "published": "",
+    "contributors": [
+        {
+            ... // reuse the Contributor model
+        },
+        ...
+    ],
+    "image": {
+        "thumbnail": {
+            ... // reuse the DigitalLocation model
+        },
+        "caption": ""
+    },
+    "partOf": [
+        {
+            ... // reuse the partOf model for series
+        }
+    ]
+}
+```
+
+```http
+GET /articles?query=foo&filters=bar&sort=baz
+{
+    "type": "ResultList",
+    "results": [
+        {
+            ... // as above
+        },
+        ...
+    ],
+    "pageSize": 100,
+    "totalResults": 1000,
+    "totalPages": 10,
+    "nextPage": "/articles?query=foo&filters=bar&sort=baz?page=2",
+}
+```
+
+#### Filters
+
+- date
+- contributor
+- series
+- contentType
+
+#### Sort
+
+- date
+- relevance
+
+### `/exhibitions`
+
+#### Example response
+
+```http
+GET /exhibitions/{id}
+{
+    "id": "",
+    "format": "",
+    "title": "",
+    "status": "",
+    "instantiations": [
+        {
+            "start": "", // ISO 8601 date only (no time)
+            "end": "", // ISO 8601 date only (no time)
+            "place": "",
+            "guide": {
+                ... // reuse the DigitalLocation model
+            }
+        }
+    ],
+    "contributors": [ // behind ?includes=contributors
+        {
+            ... // reuse the Contributor model
+        },
+        ...
+    ],
+    "image": {
+        "thumbnail": {
+            ... // reuse the DigitalLocation model
+        },
+        "caption": ""
+    },
+    "seasons": [
+        "",
+        ...
+    ],
+}
+```
+
+```http
+GET /exhibitions?query=foo&filters=bar&sort=baz
+{
+    "type": "ResultList",
+    "results": [
+        {
+            ... // as above
+        },
+        ...
+    ],
+    "pageSize": 100,
+    "totalResults": 1000,
+    "totalPages": 10,
+    "nextPage": "/exhibitions?query=foo&filters=bar&sort=baz?page=2",
+}
+```
+
+#### Filter
+
+- date
+- status
+- place
+- contributor
+- format (ie permanent, exhibition, installation)
+- season
+
+#### Sort
+
+- date
+- relevance
```

Comment:
Which is the default?

Reply:
That's interesting, because without a query you'd think it would be date, but with one, relevance. I'm guessing the date would be highly considered here in terms of relevance @harrisonpim ?

Reply:
We don't have that behaviour elsewhere though, changing the sort order based on query parameters. I'm thinking it might have to be relevance by default? It is a search tool after all and `/whats-on` still exists for ordering by date?

cc @jtweed

Reply:
I think there's also an argument for a blend ie scoring which considers recency as making content more relevant

Reply:
> I think there's also an argument for a blend ie scoring which considers recency as making content more relevant

yeah, all of the discussions we've had so far (and the tests i'm doing at the moment) have included date as secondary relevance feature by default. are there any situations where you wouldn't want to see events/exhibitions results which take the date into account?

Reply:
Seems unlikely really, so long as the date doesn't obliterate other aspects of relevance!

Reply:
Elsewhere the default order is undefined. I think there is an argument for the default order to be "useful" for listings etc, but I don't think it make sense to worry about that until and if we are using these endpoints to generate section pages. Even at that point there are arguments that it is better to be specific, as the order required might be different on different pages or we might want to AB test different algorithmic orders. So I'd go with undefined as the easiest option, then relevance if it's a full text query.

Reply:
Just to check what we mean by undefined? The current stories search is currently ordered by latest published by default but we remove it from the URL query as it's the default order. Same for works and "relevance".

Reply:
Ah sorry if we meant straight from the API, I was still under the impression it was relevance by default?

Reply:
@harrisonpim closing this after our conversation re sorting - feel free to reopen but I think you've taken it in for the newer versions of the files.